Resources/Public/JavaScript/IconElement.js

```
@@ -0,0 +1,92 @@
+        line-height: 32px;
+        color: light-dark(var(--bs-body-color), var(--typo3-input-color));
+    }
+
+    .close.svelte-df5k9v {
+        color: #000 !important;
+    }
+`};function Je(e,t){nn(new.target),kt(t,!0,Je),sn(e,es);let[r,n]=mn(),o=()=>(rn(Ht,"iconStore"),dn(Ht,"$iconStore",r)),i=mt(t,"itemFormElName",7),s=mt(t,"itemFormElValue",7),f=mt(t,"wizardConfig",7),l=mt(t,"currentIconJson",7),a=ae(null);$o(()=>{E(a,l()?JSON.parse(l()):null,!0),bn("actions-search"),bn("actions-close")});function p(){E(a,window.parent.frames.list_frame.window.SELECTED_ICON??null,!0),s(x(a)?x(a).value:""),window.parent.TYPO3.Modal.dismiss()}function _(V){V.preventDefault();let ht=JSON.parse(f()).typo3Version;Po.advanced({additionalCssClasses:["modal-bw-icon"],buttons:[{btnClass:"btn-default",name:"dismiss",icon:"actions-close",text:TYPO3.lang["button.cancel"],trigger:()=>window.parent.TYPO3.Modal.dismiss()},{btnClass:"btn-primary",name:"save",icon:"actions-document-save",text:TYPO3.lang.icon_wizard_save,trigger:p}],content:Wi`
+                <bw-icon-wizard
+                    class="w-100"
+                    itemFormElName="${i()}"
+                    wizardConfig="${f()}"></bw-icon-wizard>`,size:Po.sizes.large,title:TYPO3.lang.icon_wizard_title,staticBackdrop:!0})}var u=Qi(),v=Ue(u),y=je(v);fn(y);var T=je(y,2),A=Ue(T);let pe;var ne=Ue(A);{var Ze=V=>{var ht=Jr(),wr=Fr(ht);{var Vo=Xe=>{var Qe=Ji();Ge(()=>{Ke(Qe,"src",x(a).imgSrc),Ke(Qe,"alt",x(a).title)}),Fe(Xe,Qe)},Yo=Xe=>{var Qe=Zi();Ge(()=>_t(Qe,1,`${x(a).value??""} fontIcon`,"svelte-df5k9v")),Fe(Xe,Qe)};dr(wr,Xe=>{x(a).imgSrc?Xe(Vo):Xe(Yo,!1)})}Fe(V,ht)};dr(ne,V=>{x(a)&&V(Ze)})}He(A);var Re=je(A,2);let Ve;Re.__click=[Ki,a];var ce=Ue(Re);mr(ce,()=>o()["actions-close"]),He(Re),He(T);var $=je(T,2);$.__click=[Xi,_];var $e=Ue($);mr($e,()=>o()["actions-search"]);var vt=je($e);vt.nodeValue=` ${TYPO3.lang["wizard.button"]??""}`,He($),He(u),Ge((V,ht,wr)=>{Qr(v,`${V??""} `),Ke(y,"name",i()),pe=_t(A,1,"form-control form-control-clearable input text-center svelte-df5k9v",null,pe,ht),Ve=_t(Re,1,"close svelte-df5k9v",null,Ve,wr)},[()=>console.log(...gn("log",x(a))),()=>({"white-bg":x(a)&&!x(a).isFontIcon}),()=>({hidden:!x(a)})]),un(y,s),Fe(e,u);var Fo=Nt({get itemFormElName(){return i()},set itemFormElName(V){i(V),Ee()},get itemFormElValue(){return s()},set itemFormElValue(V){s(V),Ee()},get wizardConfig(){return f()},set wizardConfig(V){f(V),Ee()},get currentIconJson(){return l()},set currentIconJson(V){l(V),Ee()},...on()});return n(),Fo}ur(["click"]);customElements.define("bw-icon-element",xn(Je,{itemFormElName:{},itemFormElValue:{},wizardConfig:{},currentIconJson:{}},[],[],!1));export{Je as default};
```

Biome (1.9.4)

[error] This block statement doesn't serve any purpose and can be safely removed.

Standalone block statements without any block-level declarations are redundant in JavaScript and can be removed to simplify the code.
Safe fix: Remove redundant block.


(lint/complexity/noUselessLoneBlockStatements)